libpromises/evalfunction.c

```diff
@@ -509,67 +509,124 @@ static JsonElement* VarNameOrInlineToJson(EvalContext *ctx, const FnCall *fp, co
     return vardata;
 }
 
-static Rlist *GetHostsFromLastseenDB(Item *addresses, time_t horizon, bool return_address, bool return_recent)
+typedef struct {
+    char *address;
+    char *hostkey;
+    time_t lastseen;
+} HostData;
+
+static HostData *HostDataNew(const char *address, const char *hostkey, const time_t lastseen)
+{
+    HostData *data = xmalloc(sizeof(HostData));
+    data->address = xstrdup(address);
+    data->hostkey = xstrdup(hostkey);
```

Comment:
Please use `SafeStringDuplicate()` or add asserts for `address` and `hostkey` not being `NULL`.